h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -139,6 +138,29 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker) {
         gridSearch(grid);
         tryComplete();
       }
+
+      /**
+       * If export_checkpoints_dir is defined in parameters, saves the whole Grid into that
+       * folder with checkpoints.
+       *
+       * @param caller the task invoking this method (which may
+       */
+      @Override
+      public void onCompletion(CountedCompleter caller) {
+        final MP params = _hyperSpaceWalker.getParams();
+        if (params._export_checkpoints_dir != null) {
+          final String gridPath = params._export_checkpoints_dir + "/" + grid._key;
```

Comment:
since export binary is a method of Grid I thing it would be cleaner just to pass the dir and leave the attaching + grid._key for the method, it would also reduce code duplication

Reply:
Done.